### ISSUE TITLE
CopyFrom for CrateDB

### ISSUE THREAD

**Issue by Ishaq1189** [jackc/pgx#641]
Running copy_from_test.go for CrateDB throws below error
`ERROR: SQLParseException: line 1:30: mismatched input 'from' expecting {'TO', 'WHERE'} (SQLSTATE XX000)`

Please find below table schema and go code taken from copy_from_test.go to test CopyFrom functionality in CrateDB

**CrateDB table schema** 
`create table foo(
		a int2,
		b int4,
		c int8
	);`

**CopyFrom Code**
```
func  CopyFromTest() {

	conn, err := pgx.Connect(context.Background(), "postgres://crate:crate@192.168.53.221:5432/doc")
	if err != nil {
		logger.Fatalf("Unable to establish connection: %v", err)
	}

	inputRows := [][]interface{}{
		{int16(0), int32(1), int64(2)},
		{nil, nil, nil},
	}

	copyCount, err := conn.CopyFrom(context.Background(), pgx.Identifier{"foo"}, []string{"a", "b", "c"}, pgx.CopyFromRows(inputRows))
	if err != nil {
		logger.Errorf("Unexpected error for CopyFrom: %v", err)
	}
	if int(copyCount) != len(inputRows) {
		logger.Errorf("Expected CopyFrom to return %d copied rows, but got %d", len(inputRows), copyCount)
	}
}
```

**Exception** 
`ERROR: SQLParseException: line 1:30: mismatched input 'from' expecting {'TO', 'WHERE'} (SQLSTATE XX000)`

**Comment by jackc** [comment]
I don't have any first hand experience with CrateDB but it looks like it may implement `copy from` differently than PostgreSQL. In particular, pgx uses `from stdin` to stream the data over the wire from the client to the server. CrateDB's docs (https://crate.io/docs/crate/reference/en/latest/sql/statements/copy-from.html) do not mention that mode at all. It appears that its `copy from` is designed where the server receives a URI and fetches the data. So based on that I suspect CrateDB doesn't support this feature.

**Comment by Ishaq1189** [comment]
Thanks for the reply. Can you please suggest us the most performant way to bulk insert records if we don't have an option of copy from.

**Comment by jackc** [comment]
As I said I don't have an firsthand experience with CrateDB, but in regular PostgreSQL the next fastest insertion strategy would be using insert statements where each inserts multiple rows. Probably 500 or 1000 rows at a time would be a good amount per statement.

**Comment by robd003** [comment]
Here's a ticket tracking COPY FROM STDIN support in CrateDB https://github.com/crate/crate/issues/12952

Theoretically it'll be available around the time that version 5.4 is released
